metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -178,6 +178,29 @@ public Map<Urn, EntityResponse> getEntitiesV2(
         .collect(Collectors.toMap(Map.Entry::getKey, entry -> toEntityResponse(entry.getKey(), entry.getValue())));
   }
 
+  /**
+   * Retrieves the versioned aspects for the given set of urns as dynamic aspect objects
+   * (Without having to define union objects)
+   *
+   * @param entityName name of the entity to fetch
+   * @param urn urn to fetch
+   * @param aspectName aspect to fetch
+   * @param version version to fetch
+   * @return a map of {@link Urn} to {@link Entity} object
+   */
+  @Nullable
+  public EntityResponse getEntityVersionedAspectV2(
```

Comment:
nice! 

Should we just name this

getVersionedAspectV2?

Or is entity name important..? If so, why?

Reply:
I kept the entity name only because it's reflective of the service it's under since we do currently have some endpoints under "aspect." This one returns a full EntityResponse with context around the top level Urn etc. rather than just the pure aspect. Happy to change the naming if we think it's just confusing rather than helping that understanding.